### PR TITLE
Doc: Prepare for 12.0-beta1 release

### PR DESCRIPTION
**Pull request by TrueBrain**
NOTE: currently blocked while waiting for the last few PRs to be approved/merged.

## Motivation / Problem

Completely breaking a 10+ year old tradition, preparing for 12.0.

There are two things that the Steam reviews make clear:
1) our tutorial sucks (which tutorial?)
2) networking is very difficult for many people

The first we haven't addressed yet, but the second has been addressed in master now. To not let people wait another 6 months, lets make an in-between release.

## Description

As usual, an update to the changelog. As that is really all there is to "preparing" for a release :)

Why 12.0 you ask? From the commit message:
```
One question that keeps popping up: "when do we release 2.0?".
NewGRF will force that at least 1.16 will be 2.0, but to not wait
for this, let's drop the "1." and be for ever done with that
conversation.

We are following in the footstep of giants here.
```

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Comment by James103** [comment]
Builds are failing because the `compat_*.nut` files are expecting OpenTTD to be version 1.12.x, instead of the new version of 12.x.

**Comment by glx22** [comment]
Missing https://github.com/OpenTTD/OpenTTD/blob/e599f6efb4b76767f51252a6c9af41ca3fb47f48/src/ai/ai_info.cpp#L28 and https://github.com/OpenTTD/OpenTTD/blob/e599f6efb4b76767f51252a6c9af41ca3fb47f48/src/game/game_info.cpp#L26

https://github.com/OpenTTD/OpenTTD/blob/master/bin/game/CMakeLists.txt and https://github.com/OpenTTD/OpenTTD/blob/master/bin/ai/CMakeLists.txt needs an update too

**Comment by James103** [comment]
While you're at it, the milestone "1.12 Milestone" could be renamed to "12.x Milestone" for consistency with the new release version format.

File: changelog.txt [hunk 1]
```diff
@@ -1,3 +1,98 @@
...
+Add: [Script] Basic information about loaded NewGRFs for scripts (#9464)
+Add: [AI] Get the number of vehicles in a given group (#9462)
+Add: [Network] Inform network clients what game-script a server is running (#9441)
+Add: Hindi translation (#9086)
+Add: [Network] Ensure players fill in a name instead of defaulting to "Player" (#9080)
+Change: Allow pause/unpause console command in single player (#9342)
+Change: Make savegame format self-descriptive and consistent across all objects (#9322, #9335, #9338, #9339)
+Change: By default, make "unload all" leave stations empty (#9301)
+Change: Reworked the debug levels and messages for network logs (#9230, #9251)
+Change: [Emscripten] set default scrolling mode to non-pointer-locking (#9191)
```
Comment (rubidium42) [inline changelog.txt]: For consistency a capital S in `Set`.

File: changelog.txt [hunk 2]
```diff
@@ -1,3 +1,98 @@
...
+Fix: Truncating strings in settings could leave invalid UTF-8 characters (#9121)
+Fix: Many issues related to window scaling (#9087, #9219)
+Fix: Invalidate cached vehicle colourmaps when changing liveries setting (#9006)
+Fix #8981: Don't attempt to re-reserve path if already entering/entered depot (#9000)
+Fix: Missing 'Town names:' colon in map gen GUI (#8986)
+Fix: Sorting and filtering industries that produce/accept many cargoes (#8468)
+Remove: [Network] COMPANY_INFO packets (#9475)
+Remove: [Network] A server can no longer set a limit to the amount of spectators allowed (#9466)
+Remove: Arbitrary limit on number of statically loaded NewGRFs (#9431)
+Remove: [Network] language and map-name from server information (#9070)
```
Comment (rubidium42) [inline changelog.txt]: Capital l for `Language` for consistency

**Review by rubidium42 — COMMENTED** [review]
One thing to consider is whether we intend to do a 1.11 release with many of the crash/bug fixes in this, in which case those would probably be dropped from this changelog. Or that 12.0 is going to be released way sooner than Q2 2022.

**Review by michicc — APPROVED** [review]
None of the cool kids are using x.y.z anymore. Let's become a cool kid, too 😸

**Pull request merged**